Frameworks/Starboard/EbrFile.cpp

```diff
@@ -189,27 +234,27 @@ bool EbrGetRootMapping(const char* dirName, char* dirOut, uint32_t maxLen) {
         return true;
     }
     if (_stricmp(dirName, "Cache") == 0) {
-        sprintf_s(dirOut, maxLen, "%s\\cache", g_WritableFolder);
+        sprintf_s(dirOut, maxLen, "%ls\\cache", _EbrGetBaseWriteablePath());
         mkdir(dirOut);
         return true;
     }
     if (_stricmp(dirName, "Library") == 0) {
-        sprintf_s(dirOut, maxLen, "%s\\Library", g_WritableFolder);
+        sprintf_s(dirOut, maxLen, "%ls\\Library", _EbrGetBaseWriteablePath());
         mkdir(dirOut);
```

Comment:
>mkdir [](start = 8, length = 5)

same here, error value is totally ignored.
*I know it's not your code :p*, but these are all repeats. could probably be modular and do better state checking.